src/rgw/rgw_auth.cc

```diff
@@ -210,38 +364,38 @@ void RGWRemoteAuthApplier::load_acct_info(RGWUserInfo& user_info) const      /*
 }
 
 
-/* LocalAuthApplier */
+/* rgw::auth::LocalApplier */
 /* static declaration */
-const std::string RGWLocalAuthApplier::NO_SUBUSER;
+const std::string rgw::auth::LocalApplier::NO_SUBUSER;
 
-uint32_t RGWLocalAuthApplier::get_perms_from_aclspec(const aclspec_t& aclspec) const
+uint32_t rgw::auth::LocalApplier::get_perms_from_aclspec(const aclspec_t& aclspec) const
 {
   return rgw_perms_from_aclspec_default_strategy(user_info.user_id, aclspec);
 }
 
-bool RGWLocalAuthApplier::is_admin_of(const rgw_user& uid) const
+bool rgw::auth::LocalApplier::is_admin_of(const rgw_user& uid) const
 {
   return user_info.admin || user_info.system;
 }
 
-bool RGWLocalAuthApplier::is_owner_of(const rgw_user& uid) const
+bool rgw::auth::LocalApplier::is_owner_of(const rgw_user& uid) const
 {
   return uid == user_info.user_id;
 }
 
-void RGWLocalAuthApplier::to_str(std::ostream& out) const
+void rgw::auth::LocalApplier::to_str(std::ostream& out) const
 {
-  out << "RGWLocalAuthApplier(acct_user=" << user_info.user_id
+  out << "rgw::auth::LocalApplier(acct_user=" << user_info.user_id
       << ", acct_name=" << user_info.display_name
       << ", subuser=" << subuser
       << ", perm_mask=" << get_perm_mask()
-      << ", is_admin=" << user_info.admin << ")";
+      << ", is_admin=" << static_cast<bool>(user_info.admin) << ")";
 }
 
-uint32_t RGWLocalAuthApplier::get_perm_mask(const std::string& subuser_name,
-                                            const RGWUserInfo &uinfo) const
+uint32_t rgw::auth::LocalApplier::get_perm_mask(const std::string& subuser_name,
```

Comment:
hi @rzarzynski  can we  parse  `perm_mask` and `subuser_name` once? Caz we've a requirement that need to find subuser_name during auth phrase to impl [subuser usage log statistics](https://github.com/ceph/ceph/pull/13929) .